enhancements/microshift/kubernetes-for-industrial-edge.md

```diff
@@ -0,0 +1,435 @@
+via manufacturer imaging processes for large-scale production
+deployments. RHEL for Edge uses ostree-based images, and an [example
+workflow](https://github.com/openshift/microshift/blob/main/docs/rhel4edge_iso.md)
+for constructing a basic image is described in the MicroShift
+documentation. We are not creating a separate installer for
+MicroShift.
+
+#### Upgrading
+
+Because the operating system is based on rpm-ostree, upgrades are
+applied either by re-imaging the device, or by using the ostree update
+process and rebooting into a new ostree. The ostree images can be
+updated by downloading only the differences, which satisfies the
+requirement to minimize over-the-wire bandwidth costs of upgrades.
+
+An [example workflow illustrating
+upgrades](https://github.com/redhat-et/microshift-demos/tree/main/ostree-demo)
+is described in the MicroShift demos repository.
+
+There is no cluster-version-operator. MicroShift is not
+self-upgrading, although another agent on the host may participate in
+orchestrating upgrades.
+
+#### Configuring
+
+MicroShift reads its configuration file once on startup. Changes to
+the file will only be implemented after the application is
+restarted. Most configuration changes in production settings would be
+rolled out as new ostree images.
+
+### API Extensions
+
+MicroShift does not add any APIs to OpenShift.
+
+### Implementation Details/Notes/Constraints [optional]
+
+### Risks and Mitigations
+
+If an upgrade fails, the RHEL for Edge greenboot feature will allow
+the system to reboot back into the older ostree.
+
+Building a single binary means that secondary dependencies of
+components need to be kept compatible. This is especially problematic
+during a kubernetes rebase in Openhift, but applies to many other
+dependencies as well. We are working with the control plane team to
+find ways to mitigate this risk, including moving some code out of the
+OpenShift API server into controllers that would not depend on
+kubernetes at all.
+
+Building a different form factor downstream of OpenShift means it is
+possible for behaviors introduced in the dependencies to cause issues
+within MicroShift that are not caught upstream. After the basic rebase
+automation is working, we will be trying to construct a CI job that
+can be run on pull requests to the repositories containing MicroShift
+dependencies to test upstream code changes using the MicroShift
+end-to-end test job.
+
+The support lifetime of MicroShift is the same as the version of
+OpenShift on which it is based, therefore it will be critical to
+automate the component update process as much as possible to ensure
+that MicroShift stays as closely up to date as it can. The team is
+developing a [rebase
+script](https://github.com/openshift/microshift/blob/main/scripts/rebase.sh)
+which will be used by an automated job to propose update pull
+requests. We anticipate periods of time when OpenShift is itself
```

Comment:
clarify if this is during development of new releases (upcoming minors) or also during support windows of existing releases?  for existing releases, the end-user releasable artifacts must always match.

Reply:
We do rebase kube in z-stream releases of OCP, but I don't know if those rebases introduce new or different dependencies. @soltysh can you help answer that?

Reply:
I added an open question for this.